subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultCachingTransformationWorkspaceProvider.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.collect.ImmutableList;
+import org.gradle.internal.Try;
+import org.gradle.internal.UncheckedException;
+import org.gradle.internal.execution.history.ExecutionHistoryStore;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.File;
+import java.util.concurrent.ExecutionException;
+
+@ThreadSafe
+public abstract class DefaultCachingTransformationWorkspaceProvider implements CachingTransformationWorkspaceProvider {
```

Comment:
My understanding is that we use the `Default` prefix for concrete types that are to be used as the sole implementation of some interface, or at least for the majority of cases, with some specialized types taking up the rest. Therefore we shouldn't extend a `Default` implementation, nor should it be abstarct.

Reply:
Shall we call this one `AbstractCachingTransformationWorkspaceProvider` instead?

Reply:
Yes please.